waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBoss.js

```diff
@@ -1,9 +1,17 @@
 import PgBoss from 'pg-boss'
 import config from '../../../config.js'
 
-export const boss = new PgBoss({ connectionString: config.databaseUrl })
+// Add an escape hatch for advanced configuration of pg-boss.
+const pgBossNewOptions = process.env.PG_BOSS_NEW_OPTIONS || {}
+export const boss = new PgBoss({ connectionString: config.databaseUrl, ...pgBossNewOptions })
 
-// Ensure PgBoss can only be started once during a server's lifetime.
+// Allows setup code that runs before pg-boss starts to register their pg-boss functions.
+const afterStartCallbacks = []
+export function registerAfterStartCallback(callback) {
```

Comment:
🤦🏻‍♂️ Yes, this did have ordering assumptions built in that should work for our uses cases, but good catch/point. We should make it resilient to that in case the calling pattern changed! Exporting a `Promise` is a more robust method for sure. And it resolving to boss is a good idea. The only thing I can consider is if something tried to await for it that was blocking the server setup process... can think on it, but a `Promise` either way is best here. Will do, thnx!

Reply:
Updated it so we export `pgBossStarted` which people can `.then()` or `await` to get the actual `boss` handle. Great suggestion, thanks

Reply:
Awesome :)! Ah yes I know that this would work in current code since we are generating it but I keep evaluation the code from the standpoint of pretending it could change at any moment, since I am sure we will forget the intentions at some point :D. Looking good!